### ISSUE TITLE
Timeout exceeded

### ISSUE THREAD

**Issue by hexclann** [DevDavido/performance-audit-plugin#26]
An error occurred
PerformanceAudit plugin activation failed due to the following error: Node.js dependency installation failed due to the following error: The process "'/usr/bin/npm' 'install' '--quiet' '--no-progress' '--no-audit' '--force' '--only=production' '--prefix=/var/www/matomo/plugins/PerformanceAudit' 'puppeteer@^3.0' 'lighthouse@^6.0'" exceeded the timeout of 60 seconds.

I get this error while trying to install this plugin. I have already installed lighthouse package by running `sudo npm install lighthouse`

**Comment by DevDavido** [comment]
If you run `npm install --no-audit --only=production puppeteer@^3.0 lighthouse@^6.0` testwise on your server, how long does it take to perform the installation? If you try to rerun the installation, does it always timeout?

**Comment by hexclann** [comment]
I ran the command it takes about 110seconds to complete, the test server is not that powerful so that might be the issue.

I did try installing the plugin several times and it always timeout. 

With that being said, can I try installing the plugin now?

**Comment by DevDavido** [comment]
The issue with the timed out dependency installation should be fixed now with the latest release 1.1.2.
Does it work for you now with v1.1.2?

**Comment by hexclann** [comment]
Thanks for the quick fix. My vm is a 20.04 ubuntu and I had a lot of missing dependency error while trying to activate the plugin. To solve it I had to install all the dependency as mentioned [here](https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#chrome-headless-doesnt-launch-on-unix):

```
ca-certificates
fonts-liberation
libappindicator3-1
libasound2
libatk-bridge2.0-0
libatk1.0-0
libc6
libcairo2
libcups2
libdbus-1-3
libexpat1
libfontconfig1
libgbm1
libgcc1
libglib2.0-0
libgtk-3-0
libnspr4
libnss3
libpango-1.0-0
libpangocairo-1.0-0
libstdc++6
libx11-6
libx11-xcb1
libxcb1
libxcomposite1
libxcursor1
libxdamage1
libxext6
libxfixes3
libxi6
libxrandr2
libxrender1
libxss1
libxtst6
lsb-release
wget
xdg-utils
```

After installing these dependency, I could activate the plugin. I will report back to you if I find any problems

**Comment by DevDavido** [comment]
Great, glad to hear it worked out now. I added the link of the required dependencies to the [README](https://github.com/DevDavido/performance-audit-plugin#minimum-requirements) now too, so it's visible on multiple places, not only in the FAQ. If you encounter a new issue, feel free to open a new issue report.

**Comment by hexclann** [comment]
When I click on check plugin, I get this error:

```
×
ERROR: The process &quot;'/var/www/matomo/plugins/PerformanceAudit/node_modules/.bin/lighthouse' '--output=json' '--quiet' '--config-path=/tmp/phpu6mAP6' 'https://mywebsite.com/blog/' '--chrome-flags='\''--headless --disable-gpu --no-sandbox'\''' '--output-path=/var/www/matomo/plugins/PerformanceAudit/Audits/1-mobile-ee577056ee0d65ebeab240d0fe6a26c315fc2878-debug-1.json' '--emulated-form-factor=mobile'&quot; exceeded the timeout of 60 seconds. (Module: PerformanceAudit, Action: pluginCheckStart, In CLI mode: false)
```

**Comment by DevDavido** [comment]
I just released 1.1.3 to allow extending the audit timeout for each site to 5 minutes via the Matomo website settings. This should solve the audit timeout issue after enabling this setting.
